test/Conversion/HWToLLVM/convert_aggregates.mlir

```diff
@@ -72,6 +81,23 @@ func.func @convertArray(%arg0 : i1, %arg1: !hw.array<2xi32>) {
   return
 }
 
+// CHECK-LABEL: @convertConstArray
+func.func @convertConstArray(%arg0 : i1, %arg1: !hw.array<2xi32>) {
+  // CHECK:      %[[VAL_0:.*]] = llvm.mlir.constant(0 : i32) : i32
+  // CHECK-NEXT: %[[VAL_1:.*]] = llvm.mlir.constant(1 : i32) : i32
```

Comment:
We don't really have to check for this dead-code. If the pattern moves (instead of clones) constants that only the `array_create` uses, it would still be fine.